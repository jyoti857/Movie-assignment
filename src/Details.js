import * as React from 'react';
import { View, Button, StyleSheet, ScrollView, TextInput } from "react-native";
import CustomCard from './Components/Card';

export default function Details({route}){
  const {uri, movieTitle, year, type} = route.params;
  return(
    <View>
      <CustomCard 
        uri={uri} 
        movieTitle={movieTitle} 
        year = {year}
        type = {type}
        style = {styles.card}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  card: {
    width: 400
  }
})