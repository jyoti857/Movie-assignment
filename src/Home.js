import * as React from 'react';
import { View, StyleSheet, ScrollView, TextInput } from "react-native";
import CustomCard from './Components/Card';

const {useState, useEffect} = React
const url = 'http://www.omdbapi.com/?apikey=b9bd48a6&s={Query}';
function Home({navigation}){
  const [data, setData] = useState([]);
  const [dataC, setDataC] = useState([]);
  const [flag, setFlag] = useState(false);
  const [searchValue, setSearchValue] = useState()
  useEffect(() => {
    const fetchMovies = async() => {
      const response = await fetch(url);
      const result = await response.json();
      setData(result?.Search)
    }
    fetchMovies();
  }, []);
  const handleSearch = (data, elem = '') => {
    const dataClone = [...data]
    const result = dataClone && dataClone.length>0 && dataClone.filter(d => d.Title.toLowerCase().includes(elem.toLowerCase()))
    const s =  result.length > 0 ? result : [];
    setDataC(s)
    if(searchValue === '' || searchValue === undefined ){setDataC([])}
  }
  
  const handleOnTextChange = (text) => {
    setSearchValue(text)
    setFlag(!flag)
  }
  useEffect(() => {
    handleSearch(data, searchValue)
  }, [flag])
  return (
    <View style  = {styles.container}>
      <View style ={{display: 'flex', flexDirection: 'row', alignItems: 'center'}}>
        <TextInput  
          style = {styles.input} 
          placeholder='Search here by movie title' 
          value={searchValue}
          label="Email"
          onChangeText={handleOnTextChange}
        />
        
      </View>
      <ScrollView contentContainerStyle = {{flexDirection: 'row', flexWrap: 'wrap', width: 440}} >
      { 
         !searchValue ?  data && data.length > 0 && data.map((d) => <ScrollView 
         contentContainerStyle={{ backgroundColor: 'violet', flexWrap: 'wrap', width: 210, margin: 3,  display: 'flex',  flexDirection: 'row'}}
         key = {d.imdbID} horizontal showsHorizontalScrollIndicator = {false} >
            <CustomCard
              style={styles.card} 
              uri = {d.Poster} movieTitle = {d.Title} 
              onPress = {() => navigation.navigate('Details', {uri: d.Poster, movieTitle: d.Title, year: d.Year, type: d.Type})}/> 
          </ScrollView>) : dataC && dataC.length > 0 && dataC.map((dc ) => <ScrollView 
          key = {dc.imdbID}
          contentContainerStyle={{ backgroundColor: 'violet', flexWrap: 'wrap', width: 210, margin: 3,  display: 'flex',  flexDirection: 'row'}}
          horizontal showsHorizontalScrollIndicator = {false}
          >
            <CustomCard uri = {dc.Poster} movieTitle = {dc.Title}   style={styles.card} 
            onPress = {() => navigation.navigate('Details', {uri: dc.Poster, movieTitle: dc.Title,  year: dc.Year, type: dc.Type})}/> 
          </ScrollView> )
        }
      </ScrollView>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    marginTop: 40,
    marginLeft: 160, 
    marginButtom: 100,
    width: 600,
    alignSelf: 'center',
    flex:1,
    backgroundColor: 'white'
  },
  input: {
    marginTop: 30,
    height: 30,
    width: 400,
    margin: 12,
    borderRadius: 12,
    borderWidth: 1,
    paddingLeft: 4,
    color: 'red'
  },
  button: {
    width: 200,
    alignContent: 'center'
  },
  card: {
    flex: 1,
    width: 300,
    borderRadius: 20
  }
})

export default Home;