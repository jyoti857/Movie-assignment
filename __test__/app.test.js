// // import React from 'react';
import * as React from 'react';
import { View, Text } from 'react-native';
import * as renderer from 'react-test-renderer';
import Home from 'Home'
import CustomCard from '../src/Components/Card';
import Details from '../src/Details';

describe('<Card />', () => {
  // it('Movie title props', () => {
  //   expect(1).toBe(1);
  //   // const tree = renderer.create(function(){return <View><Text>eowpwe</Text></View>})
  //   const tree = renderer.create(<Home />)
  //   const testInstance = tree.root;
  //   console.log("test instance --->", testInstance)
  //   expect(testInstance.findAllByType(CustomCard).props.movieTitle).toBe('Query')
  //   console.log("tree --->", tree)
  //   // expect(tree).toMatchSnapshot();
  // })
  // it('Year props', () => {
  //   expect(1).toBe(1);
  //   const tree = renderer.create(<Home />)
  //   const testInstance = tree.root;
  //   console.log("test instance --->", testInstance)
  //   expect(testInstance.findAllByType(CustomCard).props.year).toBe('2012')
  //   console.log("tree --->", tree)
  // })
  it('Movie type props', () => {
    const tree = renderer.create(<Home />)
    const testInstance = tree.root;
    console.log("test instance --->", testInstance)
    // expect(testInstance.findAllByType(CustomCard).props.type).toBe('movie')
    console.log("tree --->", tree)
    expect(tree.toJSON()).toMatchSnapshot();
  })
  it('Card component', () => {
    const tree = renderer.create(<CustomCard />)
    const testInstance = tree.root;
    console.log("test instance --->", testInstance)
    // expect(testInstance.findAllByType(CustomCard).props.type).toBe('movie')
    console.log("tree --->", tree)
    expect(tree.toJSON()).toMatchSnapshot();
  })
  it('Details component', () => {
    const tree = renderer.create(<Details />)
    const testInstance = tree.root;
    console.log("test instance --->", testInstance)
    // expect(testInstance.findAllByType(CustomCard).props.type).toBe('movie')
    console.log("tree --->", tree)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})