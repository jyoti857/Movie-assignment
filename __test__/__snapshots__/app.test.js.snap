// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Card component 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "rgba(0, 0, 0, 0.12)",
      "borderRadius": 4,
      "elevation": 1,
      "margin": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
          },
          Object {
            "minHeight": 50,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      />
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          undefined,
          undefined,
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Title)",
            "withTheme(Card.Cover)",
            "Card.Content",
          ]
        }
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          Object {
            "paddingBottom": 16,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<Card /> Movie type props 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "white",
      "flex": 1,
      "marginButtom": 100,
      "marginLeft": 160,
      "marginTop": 40,
      "width": 600,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Search 2
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      label="Email"
      onChangeText={[Function]}
      placeholder="Search here by movie title"
      style={
        Object {
          "borderWidth": 1,
          "height": 30,
          "margin": 12,
          "marginTop": 30,
          "padding": 4,
          "width": 500,
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<Card /> works 1`] = `null`;
